whitenoise/middleware.py

```diff
@@ -168,3 +181,45 @@ def get_static_url(self, name):
             return decode_if_byte_string(staticfiles_storage.url(name))
         except ValueError:
             return None
+
+        
+
+        
+FILE_FORMATS_WITH_WEBP_SUBSTITION = ['.jpg', '.png', '.jpeg']
+
+class WebPWhiteNoiseMiddleware(WhiteNoiseMiddleware):
+
+    def process_request(self, request):
+        if self.autorefresh:
+            static_file = self.find_file(request.path_info)
+        else:
+            static_file = self.files.get(request.path_info)
+
+        if static_file is not None:
+
+            filename, ext = os.path.splitext(request.path_info)
+
+            if ext in FILE_FORMATS_WITH_WEBP_SUBSTITION:
+                if 'image/webp' in request.headers.get('Accept', ''):
```

Comment:
`image/webp;q=0` could be in the header and mean that the format is not acceptable.